File: PianoForte/Utilities/FormatManager.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace PianoForte.Utilities
{
    public class FormatManager
    {
        public static string toDisplayedPhoneNumber(string input)
        {
            string displayedPhoneNumber = "";

            if (input != "")
            {
                string part1 = "";
                string part2 = "";
                string part3 = "";

                input = input.Replace("-", "");

                int inputLength = input.Length;
                if ((inputLength == 9) && (input.Substring(0, 2) == "02"))
                {
                    part1 = input.Substring(0, 2);
                    part2 = input.Substring(2, 3);
                    part3 = input.Substring(5, inputLength - 5);
                }
                else if ((inputLength == 9) || (inputLength == 10))
                {
                    part1 = input.Substring(0, 3);
                    part2 = input.Substring(3, 3);
                    part3 = input.Substring(6, inputLength - 6);
                }

                if ((part1 != "") && (part2 != "") && (part3 != ""))
                {
                    displayedPhoneNumber = part1 + "-" + part2 + "-" + part3;
                }                
            }

            return displayedPhoneNumber;
        }
    }
}

File: Web Application/PianoForte/Web/App/js/utilities/FormatManager.js
'use strict';

goog.provide('PianoForte.Utilities.FormatManager');

PianoForte.Utilities.FormatManager = function(ValidationManager) {
    return {
        formatPhoneNumber: function(input) {
            if (input == null) { return input; };

            var formattedPhoneNumber = input;
            var isPhoneNumber = ValidationManager.isPhoneNumber(input);

            if ((input !== '') && (isPhoneNumber === true)) {
                var part1;
                var part2;
                var part3;

                input = input.replace(/-/g, '');

                var inputLength = input.length;
                if ((inputLength === 9) && (input.substring(0, 2) === '02')) {
                    part1 = input.substring(0, 2);
                    part2 = input.substring(2, 5);
                    part3 = input.substring(5, inputLength);
                } else {
                    part1 = input.substring(0, 3);
                    part2 = input.substring(3, 6);
                    part3 = input.substring(6, inputLength);                    
                }

                if ((part1 !== '') && (part1 !== '') && (part1 !== '')) {
                    formattedPhoneNumber = part1 + '-' + part2 + '-' + part3;
                }
            }

            return formattedPhoneNumber;
        },

        unformatPhoneNumber: function(input) {
            if (input == null) { return input; };
            
            var unformattedPhoneNumber = input;

            if (input !== '') {
                unformattedPhoneNumber = input.replace(/-/g, '');
            }

            return unformattedPhoneNumber;
        },

        formatNumber: function(input) {
            return accounting.formatNumber(input);
        },

        unformatNumber: function(input) {
            return accounting.unformat(input);
        }
    };
};